src/oop/user_defined_classes/hard/q5/MainTest.java

```diff
@@ -0,0 +1,109 @@
+                String.format(incorrectModifierMessage, doubleAttributeName2, testClassName));
+    }
+
+    @Test
+    public void phoneClassHasRequiredConstructor() {
+        Class<?>[] arguments = {
+                double.class, double.class
+        };
+        assertTrue(classInstance.hasConstructor(arguments),
+                String.format("Your %s class is missing a required constructor.", testClassName));
+        assertTrue(classInstance.hasConstructor(arguments, new String[]{"public"}),
+                String.format("Your %s class constructor has the incorrect visibility modifier.", testClassName));
+    }
+
+    private static Stream<Arguments> constructorInputProvider() {
+        return Stream.of(
+                Arguments.of(0.0, 1.2),
+                Arguments.of(5.42, 11.11),
+                Arguments.of(589.6, 432.9)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("constructorInputProvider")
+    public void phoneConstructorInitializesAttributesCorrectly(double attribute1, double attribute2) throws Throwable {
+        Object[][] arguments = {
+                {attribute1, double.class},
+                {attribute2, double.class}
+        };
+        Object instance = classInstance.createInstance(arguments);
+        String incorrectAttributeInstantiationMessage =
+                "Your " + testClassName + " constructor does not correctly initialize the object.";
+        _assertEquals(attribute1, classInstance.getFieldValue(instance, doubleAttributeName1),
+                incorrectAttributeInstantiationMessage);
+        _assertEquals(attribute2, classInstance.getFieldValue(instance, doubleAttributeName2),
+                incorrectAttributeInstantiationMessage);
+    }
+
+    private static Stream<Arguments> methodInputProvider() {
+        return Stream.of(
+                Arguments.of(0, 0),
+                Arguments.of(1, 0),
+                Arguments.of(0, 2),
+                Arguments.of(23.6, 8.1),
+                Arguments.of(174.429, 482.4893)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("methodInputProvider")
+    public void correctCalcScreenSizeMethod(double attribute1, double attribute2) throws Throwable {
+        Object[][] arguments = {
+                {attribute1, double.class},
+                {attribute2, double.class}
+        };
+        double diag = Math.sqrt(Math.pow(attribute1, 2) + Math.pow(attribute2, 2));
+        Object instance = classInstance.createInstance(arguments);
+        assertTrue(classInstance.hasMethod(methodName, null, double.class, new String[]{"public"}),
+                "Your " + testClassName + " class is missing the method " + methodName + " or it does not have the correct return type or visibility modifier.");
+        Object methodOutput = classInstance.callMethod(methodName, instance, null);
+        _assertEquals(diag, methodOutput,
+                "Your " + methodName + " method does not produce the correct output.");
```

Comment:
You should use the assertEquals with the delta here as you are comparing doubles

Reply:
Also, I would say that it does correctly calculate the size of the diagonal, not that it does not produce the correct output.